File: Gate Questions Python/Codes/src/EC2010,12.py
# Define logic gate functions
def AND(x, y):
    return x & y

def OR(x, y):
    return x | y

def NOR(x, y):
    return int(not (x | y))

# Define the circuit output function
def logic_circuit_output(A, B, C):
    and1 = AND(A, B)
    and2 = AND(B, C)
    or1 = OR(and1, and2)
    F = NOR(or1, C)
    return F

# Define all options
options = {
    "A": (1, 1, 0),
    "B": (1, 0, 0),
    "C": (0, 1, 0),
    "D": (0, 0, 1),
}

# Check each option
print("Verifying each option for output F = 1\n")
for key, (A, B, C) in options.items():
    result = logic_circuit_output(A, B, C)
    print(f"Option {key}: A={A}, B={B}, C={C} -> F = {result} {' Valid' if result == 1 else ' Invalid'}")

# Final answer
print("\nCorrect Option: C (A = 0, B = 1, C = 0)")

File: GATE QUESTIONS/Codes/Src/EC2010,12.c
#include <stdio.h>

// Logic gate functions
int OR(int a, int b) { return a | b; }
int NOR(int a, int b) { return !(a | b); }

int main() {
    int A, B, C, F;

    printf("Checking all input combinations for F = (NOT(A OR B)) OR C:\n\n");
    printf(" A B C | F\n");
    printf("-------|---\n");

    for (A = 0; A <= 1; A++) {
        for (B = 0; B <= 1; B++) {
            for (C = 0; C <= 1; C++) {
                int or_ab = OR(A, B);
                int z = !or_ab;     // NOR of same inputs = NOT(OR)
                F = OR(z, C);

                printf(" %d %d %d | %d\n", A, B, C, F);

                // Print the matching input
                if (F == 1) {
                    printf(">> Valid Input for F = 1: A = %d, B = %d, C = %d\n\n", A, B, C);
                }
            }
        }
    }
      printf("\nAnswer: Option (D)\n");

    return 0;
}

